### ISSUE TITLE
won't compile on macos monterey

### ISSUE THREAD

**Issue by Nevensky** [mcg1969/vecLibFort#8]
--->  Fetching archive for vecLibFort
--->  Attempting to fetch vecLibFort-0.4.2_0.darwin_21.x86_64.tbz2 from https://packages.macports.org/vecLibFort
--->  Attempting to fetch vecLibFort-0.4.2_0.darwin_21.x86_64.tbz2 from https://fra.de.packages.macports.org/vecLibFort
--->  Attempting to fetch vecLibFort-0.4.2_0.darwin_21.x86_64.tbz2 from https://cph.dk.packages.macports.org/vecLibFort
--->  Building vecLibFort
Error: Failed to build vecLibFort: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_devel_vecLibFort/vecLibFort/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is
a bug.
Error: Processing of port vecLibFort failed

**Comment by Amazingmagic** [comment]
I meet this problem, too.

**Comment by aiwen012** [comment]
same problem here when installing quantum espresso

**Comment by denny2001** [comment]
I met the same problem. Here's the log: [main.log](https://github.com/mcg1969/vecLibFort/files/7525360/main.log)

**Comment by RubenvanHooidonk** [comment]
Same problem here too
[vecLibFort.log](https://github.com/mcg1969/vecLibFort/files/7532991/vecLibFort.log)

**Comment by jserio1869** [comment]
Same problem, just jumping in.

**Comment by ghhu** [comment]
same here.

**Comment by jserio1869** [comment]
No progress on GitHub or on the macports open ticket. This should be escalated as it stops Octave installation from proceeding on Monterey.

**Comment by mcg1969** [comment]
Someone did reach out to me. I haven't even looked at this repo in years, and am frankly amazed (and heartened) that it is still proving useful to people. Unfortunately I will be unable to develop a solution to this problem but I am more than happy to review and accept a pull request.

**Comment by rshuston** [comment]
The problem isn't necessarily Monterey, but rather it's due to changes that Apple made to the `vecLib.framework` in the MacSDK12.x.sdk SDK folder, which was brought in with Xcode 13.2 (might even be Xcode 13, but I don't know). So the problem also exists on Big Sur if you have Xcode 13.2 installed.

If you examine
```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/vecLib.h
```
you'll see a new entry at line 50:
```
#include <vecLib/fortran_blas.h>
```
First of all, this line is unguarded with an "#if defined" block, so it brings in `fortran_blas.h` no matter what. Looking at the contents of `fortran_blas.h` reveals the definitions for all of the prototypes that `vecLibFort` is trying to wrap.

If you install the "Command Line Tools for Xcode 13.2" download from Apple, you'll find that it installs SDKs for both MacOSX11 and MacOSX12:
```
/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk  -> MacOSX11.3.sdk
/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk  -> MacOSX12.1.sdk
```
Examining the `vecLib.h` file in the `MacOSX11.sdk` tree shows that it's the older version that does not include `fortran_blas.h`. So, adding
```
-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk
```
to the `CFLAGS` variable in the vecLibFort `Makefile` allows the project to build the way is used to.

I'm not sure how to yet handle the Apple change, and also if the new Apple change even alleviates the need for vecLibFort entirely, but this modification may help people who need to get the library to build.

**Comment by rshuston** [comment]
I was able to fork the vecLibFort project and add a local instance of the MacOSX11.sdk version of the `vecLIb.h` file, and everything seems to compile and the `make check` also appears to work. I've submitted a PR:

https://github.com/mcg1969/vecLibFort/pull/9

**Comment by mcg1969** [comment]
Wonderful discovery @rshuston, thank you!

**Comment by rshuston** [comment]
You're very welcome. I've updated the associated MacPorts ticket with the proposed workaround that's in my PR to get the port to build for the time being.

https://trac.macports.org/ticket/63717#comment:20

**Comment by jserio1869** [comment]
Thanks @rshuston for looking at this issue! I will stay tuned on the MacPorts ticket.

**Comment by mcg1969** [comment]
Fixed by #9, which we will tag as 0.4.3
